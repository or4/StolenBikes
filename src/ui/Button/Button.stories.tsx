import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Button } from './Button';

storiesOf('Button', module).add('Simple button', () => <Button onClick={() => {}}>Next</Button>);
