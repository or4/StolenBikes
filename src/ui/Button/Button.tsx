import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { fontSize, space } from 'styled-system';
import { theme } from '../theme';

export const StyledButton = styled.div`
    display: inline-block;
    box-sizing: border-box;
    color: rgba(0, 0, 0, 0.9);
    background-color: #ffffff;
    cursor: pointer;
    box-shadow: 3px 3px 0px rgba(0, 0, 0, 0.9);
    border: 3px solid rgba(0, 0, 0, 0.9);

    line-height: 14px;
    height: 30px;
    user-select: none;

    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 100%;

    &:active {
        background-color: rgba(0, 0, 0, 0.9);
        box-shadow: 0px 0px 0px rgba(0, 0, 0, 0.9);
        border: 3px solid transparent;
        color: #ffffff;
    }
    ${fontSize}
    ${space}
`;

interface Props {
    onClick?: () => void;
}

export class Button extends React.PureComponent<Props> {
    public render() {
        const { children, onClick, ...rest } = this.props;

        const button = (
            // @ts-ignore
            <StyledButton fontSize={[2]} px={5} py={2} onClick={onClick} {...rest}>
                {children}
            </StyledButton>
        );

        return <ThemeProvider theme={theme}>{button}</ThemeProvider>;
    }
}
