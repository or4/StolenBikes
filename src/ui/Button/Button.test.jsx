import React from 'react';
import { mount, shallow } from 'enzyme';
import { Button } from '../Button';

describe('MyComponent', () => {
    it('should render correctly', () => {
        const button = <Button text={'Button test'} onClick={() => {}} />;

        const mounted = mount(button);
        expect(mounted).toMatchSnapshot();
        mounted.unmount();

        const clickFn = jest.fn();

        const component = shallow(<Button id={'button-id'} onClick={clickFn} />);
        component.find('#button-id').simulate('click');
        expect(clickFn).toHaveBeenCalled();
    });
});
