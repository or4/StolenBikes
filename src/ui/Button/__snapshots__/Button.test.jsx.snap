// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<Button
  onClick={[Function]}
>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "buttonBackground": "rgba(0, 0, 0, 0.9)",
        },
        "fontSizes": Array [
          12,
          14,
          16,
          18,
          24,
          32,
          36,
          72,
          96,
        ],
        "space": Array [
          0,
          3,
          5,
          8,
          10,
          15,
          25,
        ],
      }
    }
  >
    <styled.div
      fontSize={
        Array [
          2,
        ]
      }
      onClick={[Function]}
      px={5}
      py={2}
    >
      <StyledComponent
        fontSize={
          Array [
            2,
          ]
        }
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "fMrHvT",
              "rules": Array [
                "
    display: inline-block;
    box-sizing: border-box;
    color: rgba(0, 0, 0, 0.9);
    background-color: #ffffff;
    cursor: pointer;
    box-shadow: 3px 3px 0px rgba(0, 0, 0, 0.9);
    border: 3px solid rgba(0, 0, 0, 0.9);

    line-height: 14px;
    height: 30px;
    user-select: none;

    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 100%;

    &:active {
        background-color: rgba(0, 0, 0, 0.9);
        box-shadow: 0px 0px 0px rgba(0, 0, 0, 0.9);
        border: 3px solid transparent;
        color: #ffffff;
    }
    ",
                [Function],
                "
    ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
        px={5}
        py={2}
      >
        <div
          className="sc-bdVaJa fMrHvT"
          fontSize={
            Array [
              2,
            ]
          }
          onClick={[Function]}
        >
          Next page
        </div>
      </StyledComponent>
    </styled.div>
  </ThemeProvider>
</Button>
`;
